Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -269,13 +269,89 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
         )
       }
 
-    Signal.zip(project, baseReward, refTag)
+    let calculatedShippingTotal = Signal.combineLatest(
+      self.shippingRuleSelectedProperty.signal.skipNil(),
+      selectedRewards,
+      selectedQuantities
+    )
+    .map { shippingRule, rewards, selectedQuantities -> Double in
+      rewards.reduce(0.0) { total, reward in
+        guard reward.shipping.enabled else { return total }
+
+        let shippingCostForReward = reward.shippingRule(matching: shippingRule)?.cost ?? 0
+
+        let totalShippingForReward = shippingCostForReward
+          .multiplyingCurrency(Double(selectedQuantities[reward.id] ?? 0))
+
+        return total.addingCurrency(totalShippingForReward)
+      }
+    }
```

Comment:
A good idea here would be to re-factor these variables one block at a time. So this first one is essentially identical here and in the `PledgeViewModel`. Encapsulate this logic into one method and apply it to both view models.